esiclient/v1/node_console.py

```diff
@@ -0,0 +1,60 @@
+#   Licensed under the Apache License, Version 2.0 (the "License"); you may
+#   not use this file except in compliance with the License. You may obtain
+#   a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#   WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#   License for the specific language governing permissions and limitations
+#   under the License.
+
+import json
+import logging
+import os
+
+from osc_lib.command import command
+from osc_lib import exceptions
+from osc_lib.i18n import _
+
+
+class NodeConsoleConnect(command.Command):
+    """Connect the node console"""
+
+    log = logging.getLogger(__name__ + ".NodeConsoleConnect")
+
+    def get_parser(self, prog_name):
+        parser = super(NodeConsoleConnect, self).get_parser(prog_name)
+        parser.add_argument(
+            "node",
+            metavar="<node>",
+            help=_("node"))
+
+        return parser
+
+    def take_action(self, parsed_args):
+        self.log.debug("take_action(%s)", parsed_args)
+
+        node = parsed_args.node
+
+        ironic_client = self.app.client_manager.baremetal
+
+        try:
+            output = ironic_client.node.get_console(node)
+            if output is None:
+                raise
+
+        except Exception:
```

Comment:
If the Exception you're catching is a specific one, it's best to catch the exact one.

Reply:
Its a generic exception message! We are just catching the exception from whatever it's returning from console show.

Reply:
It's not `ironicclient.common.apiclient.exceptions.NotFound`?

Reply:
Actually, after thinking about it a bit more - I think it's fine to simply not have a try/catch block here. Then the original Ironic "node not found" exception bubbles up, and that's perfectly comprehensible to the end user.

Reply:
Used the specific exception!